decidim/decidim
# Fix newsletters unwanted CSS and 404 page on preview

#### :tophat: What? Why?
When using the image with button text and the organization has a primary color defined, the "view on web" template currently shows the CSS as plain text at the beginning of the message.

This is a persistent issue that has been fixed a couple of times before in different situations, see the related issues.

Another issue this fixes is the 404 links in the newsletter previews.

#### :pushpin: Related Issues
- Fixes #8690
- Fixes #8727
- Related to #6876
- Related to #7018

#### Testing
- Configure a primary color for the organization
- Make sure at least one user is subscribed to the newsletters
- Create a new newsletter using the "image with button" template
- In the preview section, try clicking the "see on web" or "unsubscribe" links, they now point to `#` which means they open the same page (I found this better than disabling the links as these links are coming from the translations and I did not want to add any JS to the newsletter template views)
- Send it
- From the email, click the "View on web" link
- See that the extra CSS is no longer shown as plain text and the button is in correct color
